### DISCUSSION TITLE
Why tokio does not implement itself from future crates like forward, next?

### DISCUSSION THREAD

**Discussion by doowonee** [tokio-rs/tokio#4909] [Q&A] [answered]
Sorry for my english 

It seems like if I want to use Stream (trait from future crate) I should use `tokio-stream` to create stream from a channel. But I wonder why tokio didn't implement itself `forward()` or `next()` like support by `future-util`?

**Comment by Darksonn** [comment, marked as answer]
All types in Tokio that can be converted to a `Stream` using the `tokio-stream` crate have a method that does the same as what `.next()` does when converted to a `Stream`. However, it will usually have a different name — for example, the method is called `recv` on `mpsc::Receiver`.

**Reply by doowonee** [reply]
That means `recv()` is equivalent with `next()`? How about `forward()`?

**Reply by Darksonn** [reply]
Yes, `.recv()` is equivalent to `.next()` on an `mpsc::Receiver`.

As for the other methods, we do not provide them. You are expected to use `tokio-stream` if you need them.  Regarding `forward()` in particular, that function requires use of the `Sink` trait which Tokio does not have access to for the same reasons that Tokio does not have access to the `Stream` trait.

**Reply by doowonee** [reply]
Thanks for the explanations! It was so useful!
